apps/app/src/views/routes/Settings/components/templates/TemplateForm/index.tsx

```diff
@@ -1,155 +1,236 @@
 import {
+  Checkbox,
   FormControl,
+  FormErrorMessage,
   FormLabel,
   HStack,
-  Text,
-  Checkbox,
-  VStack,
-  FormErrorMessage,
+  Heading,
+  Input,
+  Modal,
+  ModalBody,
+  ModalContent,
   ModalFooter,
+  ModalOverlay,
+  Text,
   Textarea,
-  Heading,
-  useToast,
-  Input
+  VStack,
+  useToast
 } from '@chakra-ui/react';
 
-import { ChangeEvent, RefObject, useEffect, useState } from 'react';
-import { Field, Formik } from 'formik';
+import { Field, Formik, FormikConfig } from 'formik';
+import { ChangeEvent, RefObject, useCallback, useEffect, useState } from 'react';
 
-import { OptionalText } from '../../../../../components/OptionalText';
 import { DispenseUnitSelect } from '../../../../../components/DispenseUnitSelect';
+import { OptionalText } from '../../../../../components/OptionalText';
 import { CtaButtons } from './components/CtaButtons';
-import { Treatment } from './components/Treatment';
 import { DaysSupply } from './components/DaysSupply';
-import { RefillsInput } from './components/RefillsInput';
 import { DispenseQuantity } from './components/DispenseQuantity';
+import { RefillsInput } from './components/RefillsInput';
+import { Treatment } from './components/Treatment';
 
-import { TEMPLATE_INITIAL_VALUES, TEMPLATE_SCHEMA } from './utils';
+import { types, usePhoton } from '@photonhealth/react';
+import { CATALOG_TREATMENTS_FIELDS } from 'apps/app/src/model/fragments';
+import { TEMPLATE_INITIAL_VALUES, TEMPLATE_SCHEMA, TemplateSchemaType } from './utils';
+
+type PrescriptionTemplate = types.PrescriptionTemplate;
+
+interface CreateRxTemplateVariables {
+  catalogId: string;
+  treatmentId: string;
+  dispenseAsWritten: boolean | undefined;
```

Comment:
more curious, you like this way over the `?:` shorthard? your way is definitely more legible when you skim

Reply:
they mean different things, and its actually valuable in a React context. 

```tsx
interface Props1 {
   a?: string
}

interface Props2 {
   a: string | undefined;
}

const Element1 = (props: Props1) => <div />;
const Element2 = (props: Props2) => <div />;

// no error because `a` is not required
const NoError = () => <Element1 />; 

// error: Property 'a' is missing in type '{}' but required in type 'Props2'.ts(2741)
const Error = () => <Element2 />; // 
```

Generally, especially for React elements that are part of a tree, I prefer writing it out because I'm less likely to miss a spot where its needed. In a library or where its a flag, its often better to use the `?:` version.